### ISSUE TITLE
Crash or Visual Corruption when running on 12th Gen Intel Graphics

### ISSUE THREAD

**Issue by AsgardXIV** [imchillin/Anamnesis#888]
Anamnesis will crash (or lag very hard before eventually crashing, or show corrupted visual features) when you select the Posing tab on any monitor being powered by 12th Gen Intel Graphics.

I suspect we'll see this more over time as those chips become more common out in the wild.

This appears to be a bug in the Intel drivers which impacts the WPF renderer (Viewport3D seems to be borked). I've reported it to intel.

Intel Bug Link: https://community.intel.com/t5/Graphics/Crash-with-UHD-770-in-WPF-applications-using-Viewport3D/m-p/1370393#M106529

The crash/error is as follows:
````
UCEERR_RENDERTHREADFAILURE (0x88980406)
System.Runtime.InteropServices.COMException (0x88980406): UCEERR_RENDERTHREADFAILURE (0x88980406)
   at System.Windows.Media.Composition.DUCE.Channel.SyncFlush()
   at System.Windows.Interop.HwndTarget.UpdateWindowSettings(Boolean enableRenderTarget, Nullable`1 channelSet)
   at System.Windows.Interop.HwndTarget.UpdateWindowPos(IntPtr lParam)
   at System.Windows.Interop.HwndTarget.HandleMessage(WindowMessage msg, IntPtr wparam, IntPtr lparam)
   at System.Windows.Interop.HwndSource.HwndTargetFilterMessage(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
````

I don't think there is anything we can do about it, other than track that post with Intel and see if they'll address it.

Users with discrete graphics card can use that to workaround it (see comment below).

**Comment by AsgardXIV** [comment]
# Workarounds

## Discrete vs Integrated Graphics
This issue only affects integrated graphics on 12th gen Intel processors.
If you have a discrete GPU (NVidia or AMD) follow the steps below to work around it.

## Desktops with Discrete Graphics 
Ensure you run Anamnesis on a monitor physically connected to your discrete graphics card.

## Laptops with Discrete Graphics
Ensure Anamnesis is running on your Discrete Graphics cards.
There's a guide on how to do this here: https://www.onmsft.com/how-to/how-to-set-per-app-graphical-performance-settings-in-windows-10

Windows 11 is basically the same.

## Only Integrated Graphics
Unfortunately there is no workaround for this if you do not have discrete GPUs

**Comment by AsgardXIV** [comment]
I got a response from Intel to say they reproduced the issue and they it's fixed in an internal driver build. No timeline on a release yet. 

Will update when I get a date or version.

**Comment by AsgardXIV** [comment]
Intel let me know and I have now confirmed that this is fixed in the latest driver release, `31.0.101.3222`.

https://www.intel.ca/content/www/ca/en/download/19344/intel-graphics-windows-dch-drivers.html

**Comment by Yuki-Codes** [comment]
Looks like intel has reintroduced some rendering issues, so am leaving this pinned.

**Comment by AsgardXIV** [comment]
They claim to have fixed it again in `31.0.101.3790` and later.
I confirm that it works again for me. Hopefully it will remain working.

Can be grabbed at the same link: https://www.intel.ca/content/www/ca/en/download/19344/intel-graphics-windows-dch-drivers.html

**Comment by davzell13** [comment]
> They claim to have fixed it again in `31.0.101.3790` and later. I confirm that it works again for me. Hopefully it will remain working.
> 
> Can be grabbed at the same link: https://www.intel.ca/content/www/ca/en/download/19344/intel-graphics-windows-dch-drivers.html

hello, i have a similiar duce Chanel error on same graphic card and with the most recent driver.
Is it possible to have the email adress of intel to write to them my problem ? 

Thanks
